### PR TITLE
General improvements

### PR DESCRIPTION
**Pull request by hacker1024**
This PR includes a number of general improvements. See the commit list for details.
Some highlights include:

- Disks are detected simultaneously and asynchronously, no longer blocking the main thread and reducing the complexity from O(n) to a theoretical O(1)
- `final` and `const` are used wherever possible, including in the `Disk` constructor
- getDisk works with all `FileSystemEntities`, and doesn't need them to actually exist

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -2,14 +2,10 @@ name: universal_disk_space
 description: parses total and available disk spaces on Windows and Linux-based systems
 version: 0.1.20
 homepage: https://github.com/joaquimguimaraes/universal_disk_space
-authors:
-  - Gil Nóbrega <jgilfn@gmail.com>
```
Comment (vleeuwenmenno) [inline pubspec.yaml:6]: Removal of authors on a PR is not a good way to impress people 🤣

Comment (hacker1024) [reply pubspec.yaml:6]: 🤷🏻 It is deprecated: https://dart.dev/tools/pub/pubspec#authorauthors

Comment (hacker1024) [reply pubspec.yaml:6]: This PR is open to maintainers, so the commit can be reverted if it's an issue.

Comment (vleeuwenmenno) [reply pubspec.yaml:6]: Doesn't seem like an issue to me, was just wondering why so. But that clarifies the situation.

**Comment by gilnobrega** [comment]
Thank you for improving to the code's structure, readability and documentation
Async disk scan will be very useful

**Pull request merged**